lib/ts/recipe/totp/api/implementation.ts

```diff
@@ -14,82 +14,84 @@
  */
 
 import { APIInterface } from "../types";
-import STError from "../error";
 import TotpRecipe from "../recipe";
+import SessionError from "../../session/error";
+// import { MfaClaim, completeFactorInSession } from '../../mfa';
 
 export default function getAPIImplementation(): APIInterface {
     return {
-        createDevicePOST: async function (input) {
-            const { session, options, ...rest } = input;
-            let userIdentifierInfo = undefined;
+        createDevicePOST: async function ({ session, options, deviceName, userContext }) {
+            let userIdentifierInfo: string | undefined = undefined;
             const emailOrPhoneInfo = await TotpRecipe.getInstanceOrThrowError().getUserIdentifierInfoForUserId(
                 session.getUserId(),
-                input.userContext
+                userContext
             );
             if (emailOrPhoneInfo.status === "OK") {
                 userIdentifierInfo = emailOrPhoneInfo.info;
+            } else if (emailOrPhoneInfo.status === "UNKNOWN_USER_ID_ERROR") {
+                throw new SessionError({
+                    type: SessionError.UNAUTHORISED,
+                    message: "Unknown User ID provided",
+                });
             }
-            const args = { ...rest, userId: session.getUserId(), userIdentifierInfo };
-            let response = await input.options.recipeImplementation.createDevice(args);
-
-            return response;
-        },
-
-        verifyCodePOST: async function (input) {
-            const { session, options, ...rest } = input;
-            const args = { ...rest, userId: session.getUserId() };
-            let response = await input.options.recipeImplementation.verifyCode(args);
 
-            if (response.status === "TOTP_NOT_ENABLED_ERROR") {
-                throw new STError({
-                    type: "TOTP_NOT_ENABLED_ERROR",
-                    message: "TOTP is not enabled for this user",
+            let existingDeviceCount = 0;
+            let verifiedDeviceCount = 0;
+            if (deviceName === undefined) {
+                // We need to set the device name:
+                const listDevicesResponse = await options.recipeImplementation.listDevices({
+                    userId: session.getUserId(),
+                    userContext,
                 });
+                const devices = listDevicesResponse.status === "OK" ? listDevicesResponse.devices : [];
+                existingDeviceCount = devices.length;
+                verifiedDeviceCount = devices.filter((device) => device.verified).length;
+                deviceName = `TOTP Device ${existingDeviceCount + 1}`; // Assuming no one creates a device in the same format
             }
-            return response;
+
+            if (verifiedDeviceCount > 0) {
```

Comment:
if the deviceName is provided by the user, the verifiedDeviceCount will be 0, cause you set the value to it only if `deviceName === undefined`

Reply:
?